### PR TITLE
New LoggingNode and SensorPlayer

### PR DESCRIPTION
**Pull request by VivaanBahl**
Moved LoggingNode and SensorPlayer to the GSON library, a much nicer interface with dealing with JSON, as well as using a Reflections basis to make serializing and deserializing much easier.

### CODE REVIEW

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 1]
```diff
@@ -45,18 +43,16 @@ public static void main(String[] args)  {
      	
     	if (RobobuggyConfigFile.DATA_PLAY_BACK) {
     		//Play back mode enabled
-    		final SensorPlayer sp = new SensorPlayer("logs/2016-01-26-00-00-53/sensors.txt");
+    		final SensorPlayer sp = new SensorPlayer("logs/sensor.txt");
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: why do  you wanted it changed to sensor.txt

**Comment by VivaanBahl** [comment]
That was an artifact of testing that I need to change

**Comment by VivaanBahl** [comment]
Note that this changes the schema of the log files. I can make it backwards compatible if necessary

**Comment by TrevorDecker** [comment]
has this been tested?

**Comment by VivaanBahl** [comment]
functionally (with small sample sizes). I have yet to test it with full performance

**Comment by TrevorDecker** [comment]
also can we have a folder for each log

**Comment by TrevorDecker** [comment]
I am adding video recording and playback so I want to give each log its own folder like before

**Comment by VasuAgrawal** [comment]
Backward compatibility would be nice, or at the very least make some tool which changes logs to the newer format. I'd prefer the former though.

**Comment by VasuAgrawal** [comment]
As we're doing the logging clean up, can someone please go through all of our log files and figure out which ones are necessary and clean them up? New member task, perhaps?

**Comment by TrevorDecker** [comment]
I am approving this :)

**Pull request merged**